qadence/ml_tools/saveload.py

```diff
@@ -102,13 +113,18 @@ def write_checkpoint(
 
 
 def load_model(
-    folder: Path, model: Module, model_ckpt_name: str | Path = "", *args: Any, **kwargs: Any
+    folder: Path,
+    model: Module,
+    model_ckpt_name: str | Path = "",
+    device: str | torch.device = "cpu",
+    *args: Any,
+    **kwargs: Any,
 ) -> tuple[Module, int]:
     from qadence import QNN, QuantumModel
```

Comment:
Personally I'd expect here to be able to send `device = cuda:4`, and have the model loaded on GPU 4. 
Since we store the model as `cuda` though, that means that we need to change the logic here. 

I would suggest to pass `device` to the torch.load function, and then we'll have to sort out how to deal with the  
```
model._from_dict
```
and
```
model.load_state_dict
```
parts I guess.

Reply:
If it's not obvious how to do it we can make it a new issue, it's a new feature I suppose, although it's also a bug :p

Reply:
We could handle that by simply adding `model.to(device)` after `model._from_dict` and `model.load_state_dict` maybe? That way, if e.g. `cuda:4` is passed then the model would be found and loaded no matter in which GPU the model is saved (since we only save and look for `cuda` when loading) and then it would be moved into GPU 4

Reply:
Yeah, it should probably be sufficient.